### PR TITLE
dist: provide new docker-compose-based Grist distribution

### PR DESCRIPTION
**Pull request by jordigh**
The idea is to replace the existing dependency on Grist Omnibus and distribute this Docker Compose configuration instead both for AWS and Digital Ocean virtual machine images.

This setup has been synthesised from work done by @paulfitz and @Spoffy on Grist Omnibus and Docker Compose examples respectively.

A first draft of this code has already been reviewed informally in this PR: https://github.com/gristlabs/grist-pack/pull/2

There have been minor changes since then. Apologies for the difficulty in reviewing.

### CODE REVIEW

**Comment by paulfitz** [comment]
Hi @jordigh, can you add a README or similar sketching anything worth knowing about how to use this?

**Comment by jordigh** [comment]
Oh, at the moment it's just `./generate-secrets && docker compose up` and point your web browser to `http://grist.localhost`. You can edit between those two commands the generated `.env` file with some changes.

I'll write this up.

File: dist/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Grist Cloud Distribution
+
+This a comprehensiveversatile distribution of Grist made with Docker Compose. It
```
Comment (paulfitz) [inline dist/README.md:3]: Did you intend to make this neologism or is it a typo?

File: dist/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+docker compose up
+```
+
+This should start, in this order, Traefik, Authelia, Dex, and finally
+Grist, at the URL specified by `APP_HOME_URL` (by default, at
+`http://grist.localhost`). You should then be able to visit that URL
+in a web browser and log in to Grist.
+
+<!--TODO: let user configure the default username/password -->
+Currently, the default usernmae to log into Authelia is `test` with
```
Comment (paulfitz) [inline dist/README.md:40]: Can you emphasize the defaults a bit more. Once I saw "docker compose up" I went off to try that, and only came back here to try and find a username/password. I then missed the URL in the sentence above and used `localhost`. If you put a block with:

Default URL: http://grist.localhost
Default username: test
Default password: test

there's a better chance of the first run being pleasingly successful.

Comment (paulfitz) [inline dist/README.md:40]: *username

File: dist/bin/generate-secure-secrets [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Helper script to securely generate random secrets for Authelia.
+
+CONFIG_DIR=$(realpath $(dirname $0)/../config)
```
Comment (paulfitz) [inline dist/bin/generate-secure-secrets:5]: maybe an early `set -e` to fail immediately if anything fails? Helps when users have trouble.

**Review by paulfitz — COMMENTED** [review]
Just reviewing the README. It worked! :tada:

File: dist/bin/generate-secure-secrets (hunk 3 above)
Comment (jordigh) [reply dist/bin/generate-secure-secrets:5]: Good idea, thanks.

File: dist/README.md (hunk 2 above)
Comment (jordigh) [reply dist/README.md:40]: Oops.

Comment (jordigh) [reply dist/README.md:40]: Sure. But I don't expect this README will be much of an end-user experience in the end. I hope it won't be.

File: dist/README.md (hunk 1 above)
Comment (jordigh) [reply dist/README.md:3]: At first I thought it was versatile, then I realise maybe it wasn't, and I left the edit mistake in there.

File: dist/config/env-template
[the remark this answers is not in the document]
Comment (jordigh) [reply dist/config/env-template:1]: I plan to make them accessible via a setup script to be run upon first ssh contact.

File: dist/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+docker compose up
+```
+
+This creates the following default setup:
+
+* Grist at http://grist.localhost
+<!--TODO: let user configure the default username/password -->
+* Default username: `test`
+* Default password: `test`
+
```
[the remark this answers is not in the document]
Comment (jordigh) [reply dist/README.md:34]: Good idea, added.

**Pull request merged**